include/xgboost/data.h

```diff
@@ -35,10 +35,55 @@ enum class DataType : uint8_t {
 };
 
 enum class FeatureType : uint8_t {
-  kNumerical,
-  kCategorical
+  kIndicator = 0,
+  kQuantitive = 1,
+  kInteger = 2,
+  kFloat = 3,
+  kCategorical = 4,
```

Comment:
This change will break the serialization logic here:
https://github.com/dmlc/xgboost/blob/d711d648cb83409da00ff48e1890e1e9e3868566/src/tree/tree_model.cc#L909-L932
where the `split_types_` array is initialized to 0 (`kNumerical`).